cmd/libsnap-confine-private/string-utils.c

```diff
@@ -56,6 +56,22 @@ bool sc_endswith(const char *str, const char *suffix)
 	return strncmp(str - xlen + slen, suffix, xlen) == 0;
 }
 
+char *sc_strdup(const char *str)
```

Comment:
Considering that it calls to `die()` on failure, I'd call it `sc_must_strdup()`, similarly to `sc_must_snprintf()`.

Reply:
Virtually all functions die, I think the must thing is a useless add-on that doesn't reflect the permadeath mode of the code.

Reply:
I'm slightly agreeing with @bboozzoo here: similiar to `sc_must_sprintf`, this new function belongs to primitives that you normally don't expect to abort, so emphasizing it makes sense (also for consistency with sc_must_snprintf).

Reply:
There are many dozen functions using the sc_ prefix that all die an any error. The only one of those that uses the word _must_ is `sc_must_snprintf`. I'd really rather invent an opposite word if we want to have this sort of consistency.